index.js

```diff
@@ -82,14 +106,31 @@ function fastifyWebsocket (fastify, opts, next) {
     oldClose.call(this, cb)
   }
 
-  function wsHandle (handle, req, res) {
-    return handle.call(fastify, req[kWs], res)
+  function noHandle (conn, req) {
+    req[kWs].socket.close()
   }
 
-  function noHandle (req, res) {
-    req[kWs].socket.close()
+  function defaultErrorHandler (conn, error) {
+    // Before destroying the connection, we attach an error listener.
+    // Since we already handled the error, adding this listener prevents the ws
+    // library from emitting the error and causing an uncaughtException
+    // Reference: https://github.com/websockets/ws/blob/master/lib/stream.js#L35
+    conn.on('error', _ => {})
+    conn.destroy(error)
```

Comment:
I think we should log the error somewhere.

Reply:
Do you think using `fastify.log` is the right way to do it?

Reply:
You should really use `req.log` as it already has some details on the current requests. You might also want to attach that to the conn.

Reply:
Since the error handler doesn't _always_ have access to the `FastifyRequest`, there would be no way to use `req.log`

Reply:
Use fastify.log for now, but we should add a few TODOs / issues about logging.

Reply:
Done, I'd like to note that removing usage of the global handler would make this easier, allow all handlers to have access to `FastifyRequest`/`FastifyReply` objects, and keep things uniform.

Have you considered that possibility?

Reply:
Let's remove it. Let's add a test that verifies it all work with a wildcard route: `/*`.

Reply:
I've removed usage of the global handler (keeping `noHandle` to close connections on unregistered routes)